browser/js/common/factories/snippet-factory.js

```diff
@@ -106,12 +106,45 @@ app.factory("Snippet", function($firebaseObject, AuthService, Users, $rootScope,
             })
             updates[`/users/${snippet.team}/snippets/asManager/${snippetId}`] = null;
             updates[`/users/${snippet.owner}/snippets/asOwner/${snippetId}`] = null;
+            updates[`/users/${snippet.owner}/snippets/asCollaborator/${snippetId}`] = null;
             updates[`/users/${snippet.owner}/snippets/asTeamMember/${snippetId}`] = null;
             updates['/snippets/' + snippetId] = null;
-            return ref.update(updates);
+            console.log('setting the updates');
+            return ref.update(updates).then(function(result){
+                return result;
```

Comment:
Unnecessary return. The promise is already a promise for the result, prior to the `.then`.